spec/features/deal_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe 'Deals', type: :feature do
+  let(:user) { User.create(name: 'Test', email: 'test@example.com', password: 'password') }
+  let!(:category1) { Category.create(Name: 'Test1', icon: fixture_file_upload('test.png', 'image/png'), user:) }
+  let!(:category2) { Category.create(Name: 'Test2', icon: fixture_file_upload('test.png', 'image/png'), user:) }
+  let!(:payment1) { Deal.create(name: 'Test1', amount: 10, author: user) }
+  let!(:payment2) { Deal.create(name: 'Test2', amount: 20, author: user) }
+
+  before do
+    category1.deals << payment1
+    category2.deals << payment2
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Log In'
+    visit category_path(category1)
+  end
+
+  scenario 'displays a list of payments with their categories' do
```

Comment:
- [ ] [REQUIRED] Great job so far @sajibarafatsiddiqui however, we have a test failure :
<img width="825" alt="Screenshot 2023-06-10 at 16 34 32" src="https://github.com/sajibarafatsiddiqui/budget_app/assets/58692590/1e356a16-8f90-4f1b-9459-29e590c9a988">
The error message suggests that Capybara is unable to find an input field with the label or placeholder "Email" on the page

### Hint to resolve the issue
Replace 'user_email' with the correct attribute value associated with the email field.
``` ruby
fill_in 'Email', with: user.email
```

Reply:
<img width="611" alt="image" src="https://github.com/sajibarafatsiddiqui/budget_app/assets/39811497/23235467-b4b2-4830-b69a-7b34f737b799">
<img width="323" alt="image" src="https://github.com/sajibarafatsiddiqui/budget_app/assets/39811497/8a66b78b-af28-4c34-8e6b-57427448ba4f">

both issues have been resolved

Reply:
- Well resolved
<img width="318" alt="Screenshot 2023-06-10 at 19 38 48" src="https://github.com/sajibarafatsiddiqui/budget_app/assets/58692590/f1da2e1e-4c0f-4213-95e4-db23cc390d3a">

Reply:
- Well resolved 
<img width="391" alt="Screenshot 2023-06-10 at 19 39 56" src="https://github.com/sajibarafatsiddiqui/budget_app/assets/58692590/a042c20a-e3b1-465c-bd5e-d35f7e3022b3">